src/main/java/uk/gov/hmcts/reform/sscs/migration/service/LocationRefDataService.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.sscs.migration.service;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import java.util.Collections;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.client.RefDataApi;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+import uk.gov.hmcts.reform.sscs.model.CourtVenue;
+
+@Service
+@RequiredArgsConstructor
+public class LocationRefDataService {
+
+    private static final String SSCS_COURT_TYPE_ID = "31";
+
+    private final IdamService idamService;
+    private final RefDataApi refDataApi;
+
+    public List<CourtVenue> retrieveCourtVenues() {
+        IdamTokens idamTokens = idamService.getIdamTokens();
+
+        if (idamTokens == null
```

Comment:
@nickwest1 I think you'll like this bit :)

Reply:
hehe, yep, you're right there Waqas :)